### PR TITLE
3764 import nested thesaurus

### LINKED ISSUE #3764: CSV import of nested thesaurus values fails (8 pts)
If you CSV import an entity with a select or multiselect with the value of a nested thesaurus item, the value appears on cards (denormalized?) but when you edit the entity, it doesn't show on the selectors.

### PR DESCRIPTION
**Pull request by mfacar**
fixes #3764

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Review by fnocetti — CHANGES_REQUESTED** [review]
This LGTM. I just left a couple of not-that-important comments.
I'm proceeding to the testing now.

**Comment by fnocetti** [comment]
@mfacar Everything seems to work nicely, except for the following issue:
If in the CSV file you put the label for a thesauri group (or root thesauri) it is correctly imported, and shown in the cards. When you go to the entity view or to entity edit, the value is not rendered.
You should only be allowed to select leaf thesauri values, right?

**Comment by mfacar** [comment]
> @mfacar Everything seems to work nicely, except for the following issue: If in the CSV file you put the label for a thesauri group (or root thesauri) it is correctly imported, and shown in the cards. When you go to the entity view or to entity edit, the value is not rendered. You should only be allowed to select leaf thesauri values, right?

That's right, however probably it should be a new issue, because it's also happening at sending the value through the API. I'll take a look here and add the solution if it's straightforward.

File: app/api/entities/validation/validateMetadataField.ts [hunk 1]
```diff
@@ -52,13 +52,14 @@ const validateType = (
...
-const flattenDictionaryValues = (dictionary: ThesaurusSchema) =>
-  (dictionary.values || []).reduce<ThesaurusValueSchema[]>((flattened, v) => {
-    if (v.values?.length) {
-      return flattened.concat(v.values);
-    }
-    return flattened.concat([v]);
-  }, []);
+const compareThesaurusValue = async (property: PropertySchema, value: MetadataObjectSchema[]) => {
+  const thesaurus = await thesauris.getById(property.content);
+  const thesaurusValues = _.flatMapDeep(thesaurus?.values, tv =>
```
[the remark this answers is not in the document]
Comment (mfacar) [reply app/api/entities/validation/validateMetadataField.ts]: done

File: app/api/csv/specs/csvLoaderSelects.spec.js [hunk 2]
```diff
@@ -224,4 +117,57 @@ describe('loader', () => {
...
+      );
+
+    beforeAll(async () => {
+      actualEntities = await entities.get({
+        template: fixtureFactory.id('template'),
+        language: 'en',
+      });
+    });
+
+    it('should only add as new values those which are not nested values', async () => {
```
[the remark this answers is not in the document]
Comment (mfacar) [reply app/api/csv/specs/csvLoaderSelects.spec.js]: done

**Review by fnocetti — APPROVED** [review]
LGTM!

**Pull request merged**